components/treeview/drag-drop.md

```diff
@@ -49,13 +82,212 @@ The `OnDrop` event provides an object of type `TreeViewDropEventArgs` to its eve
 | `DestinationIndex` | `string` | The index where the drop will happen in the second component. |
 | `DestinationComponentId` | `string` | The `Id` of the second component in which the drop will happen. |
 
+## OnDragEnd Event
+
+The `OnDragEnd` event fires when a drag operation is ends. The event is triggered after the `OnDrop` Event and unlike it, `OnDragEnd` will fire even if the new location is not a Telerik component.
+
+### Event Arguments
+
+The `OnDragEnd` event handler receives as an argument an object of type `TreeViewDragEndEventArgs` that contains:
+
+| `DestinationItem` | `object` | Represents the row over which the `Item` is. You can cast this object to your model class. |
+| `DestinationTreeView` | `object` | The reference of the TreeView in which the `Item` is dropped. |
+| `DestinationIndex` | `string` | The index in the target component where the drop will happen. |
+| `DestinationComponentId` | `string` | The `Id` of the target component in which the drop will happen. |
+| `DropPosition` | `enum` | Its members allow you to determine the exact position of the dropped item relative to the position of the `DestinationItem`. |
+
 ## Examples
 
+* [Events Example](#events-example)
 * [Drag and Drop between TreeView, Grid, TreeList and Scheduler](#drag-and-drop-between-treeview-grid-treelist-and-scheduler)
 * [Flat Data](#flat-data)
 * [Hierarchical Data](#hierarchical-data)
 * [Between Different TreeViews](#between-different-treeviews)
 
+### Events Example
```

Comment:
Consider "TreeView Drag and Drop Events" for better SEO. No need to repeat "example"